packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -216,6 +216,94 @@ export interface ClusterOptions {
+   *
+   * @see https://docs.aws.amazon.com/eks/latest/userguide/cluster-endpoint.html
+   *
+   * @default EndpointAccess.publicAndPrivate()
+   */
+  readonly endpointAccess?: EndpointAccess;
+
+  /**
+   * Environment for the kubectl execution. Only relevant for kubectl enabled clusters.
+   *
+   * These variables will be injected to the handler executing kubectl/helm commands.
+   *
+   * @default - No environment.
+   */
+  readonly kubectlEnvironment?: { [key: string]: string }
+}
+
+/**
+ * Endpoint access characteristics.
+ */
+export class EndpointAccess {
+
+  /**
+   * The cluster endpoint is accessible from outside of your VPC.
+   * Worker node traffic will leave your VPC to connect to the endpoint.
+   *
+   * By default, the endpoint is exposed to all adresses. You can optionally limit the CIDR blocks that can access the public endpoint using the `PUBLIC.onlyFrom` method.
+   * If you limit access to specific CIDR blocks, you must ensure that the CIDR blocks that you
+   * specify include the addresses that worker nodes and Fargate pods (if you use them)
+   * access the public endpoint from.
+   *
+   * @param cidr The CIDR blocks.
+   */
+  public static readonly PUBLIC = new EndpointAccess(false, true);
+
+  /**
+   * The cluster endpoint is only accessible through your VPC.
+   * Worker node traffic to the endpoint will stay within your VPC.
+   */
+  public static readonly PRIVATE = new EndpointAccess(true, false);
+
+  /**
+   * The cluster endpoint is accessible from outside of your VPC.
+   * Worker node traffic to the endpoint will stay within your VPC.
+   *
+   * By default, the endpoint is exposed to all adresses. You can optionally limit the CIDR blocks that can access the public endpoint using the `PUBLIC_AND_PRIVATE.onlyFrom` method.
+   * If you limit access to specific CIDR blocks, you must ensure that the CIDR blocks that you
+   * specify include the addresses that worker nodes and Fargate pods (if you use them)
+   * access the public endpoint from.
+   *
+   * @param cidr The CIDR blocks.
+   */
+  public static readonly PUBLIC_AND_PRIVATE = new EndpointAccess(true, true);
+
+  private constructor(
+    /**
+     * Indicates if private access is enabled.
+     */
+    public readonly privateAccess: boolean,
+
+    /**
+     * Indicates if public access is enabled.
+     */
+    public readonly publicAccess: boolean,
+    /**
+     * Public access is allowed only from these CIDR blocks.
+     * An empty array means access is open to any address.
+     */
+    public readonly publicCidrs?: string[]) {
+    if (!publicAccess && publicCidrs && publicCidrs.length > 0) {
+      throw new Error('CIDR blocks can only be configured when public access is enabled');
+    }
+  }
+
+
+  /**
+   * Restrict public access to specific CIDR blocks.
+   * If public access is disabled, this method will result in an error.
+   *
+   * @param cidr CIDR blocks.
+   */
```

Comment:
Add @experimental

Reply:
Yeah? Not sure its necessary, I think the API is ok, and we aren't GAing yet anyway...